state_machine/applications/system-check/tests/sensor_rate_test.py

```diff
@@ -0,0 +1,49 @@
+import time
+try:
+    from ulab.numpy import array, median
+except ImportError:
+    from numpy import array, median
+from lib.pycubed import cubesat
+
+def sensor_rate_test(sensor, name):
+    trials = 100
+    li = [0.0] * trials
+    dt_list = array(li)
+    if name == "IMU":
+        for i in range(trials):
+            s_time = time.monotonic()
+            _ = sensor.accel
+            _ = sensor.accel
+            e_time = time.monotonic()
+            dt_list[i] = e_time - s_time
+    else:
+        for i in range(trials):
```

Comment:
This is essentially a duplicate of the code above. Could you make it a function and just pass in the particular property you want to call?

One way to do that would be to pass in a lambda

`lambda: sensor.accel`

or follow this
https://stackoverflow.com/questions/46858184/how-to-pass-setter-property-as-argument-to-a-function